crates/scheduler/src/call/platform/protocol.rs

```diff
@@ -40,17 +40,17 @@ fn read<B: Board>(mut call: SchedulerCall<B, api::read::Sig>) {
     let scheduler = call.scheduler();
     let result = try {
         match scheduler.applet.get_request() {
-            Ok(None) => Ok(false),
+            Ok(None) => false,
             Ok(Some(value)) => {
                 let mut memory = scheduler.applet.memory();
                 let len = value.len() as u32;
                 let ptr = memory.alloc(len, 1)?;
                 memory.get_mut(ptr, len)?.copy_from_slice(&value);
                 memory.get_mut(*ptr_ptr, 4)?.copy_from_slice(&ptr.to_le_bytes());
                 memory.get_mut(*len_ptr, 4)?.copy_from_slice(&len.to_le_bytes());
-                Ok(true)
+                true
             }
-            Err(e) => Err(e),
+            Err(e) => Err(e)?,
```

Comment:
This is actually not needed anymore, the whole match can be simplified by using `?` on the match expression.

```rust
match scheduler.applet.get_request()? {
```

And then, we're simply matching on an `Option`.